### PR TITLE
RavenDB-7940  Recreate indexes on the database after recreate a database

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/IndexStore.cs [hunk 1]
```diff
@@ -405,7 +405,15 @@ public Index GetIndex(string name)
...
                 {
                     var (etag, _) = await _serverStore.SendToLeaderAsync(command);
-                    await _documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(etag);
+                    try
+                    {
+                        await _documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(etag , timeoutForIndexNotification);
+
+                    }
+                    catch (TimeoutException)
+                    {
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/Indexes/IndexStore.cs:414]: It should be ignored only if we are recreating indexes

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/IndexStore.cs:414]: No, it should never be ignored. What is the scenario where that is a good thing to do?

File: src/Raven.Server/Documents/Indexes/IndexStore.cs [hunk 2]
```diff
@@ -437,9 +445,16 @@ public Index GetIndex(string name)
...
 
-                var (etag, _) = await _serverStore.SendToLeaderAsync(command);
+                var (etag, _) = await _serverStore.SendToLeaderAsync(command);      
+                try
+                {
+                    await _documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(etag, timeoutForIndexNotification);
 
-                await _documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(etag);
+                }
+                catch (TimeoutException)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/IndexStore.cs:454]: Why is ignoring the exception a good idea?

File: src/Raven.Server/Documents/Indexes/IndexStore.cs [hunk 3]
```diff
@@ -912,11 +970,18 @@ private void OpenIndexesFromRecord(PathSetting path, DatabaseRecord record)
                     OpenIndex(path, definition.Etag, indexPath, exceptions, name);
             }
 
+            if (Directory.Exists(path.FullPath) &&
+                record.AutoIndexes.Count == 0 && record.Indexes.Count == 0)
+            {
+                RecreateIndexes(path, record, exceptions);
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/IndexStore.cs:976]: No, this should NOT be called as part of the index store / document store startup.
This should be called as part of the Create Database.

File: test/FastTests/Issues/RavenDB-7940.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                {
+                    DatabaseName = store.Database,
+                    Settings = new Dictionary<string, string>
+                    {
+                        {"DataDir", path},
+                        {"RunInMemory" , false.ToString()}
+                    }
+                }));
+
+                SpinWait.SpinUntil(() =>
```
Comment (ayende) [inline test/FastTests/Issues/RavenDB-7940.cs:77]: Avoid SpinWait in tests.

**Review by ayende — CHANGES_REQUESTED** [review]

**Pull request closed without merging**